### ISSUE TITLE
Table: VirtualScroll with LazyLoad skips second page

### ISSUE THREAD

**Issue by monkey-mku** [primefaces/primeng#13299]
### Describe the bug

I updated the demo to more closely resemble our real life scenario. The number of rows only is known after data is actually loaded and gets updated as you filter the data. Also scrollHeight is flex so it adjusts to the content around it.

Now the table component is firing incorrect lazy load events. The first page gets loaded 3 times and when you start scrolling afterwards the second page gets skipped completely. Only once you scroll past that the third page gets loaded again.
I have noticed that the second page does get loaded if you immediately start scrolling after the second event for the first page gets fired.
If you filter the data the same issue occurs if you wait a few seconds before you start scrolling. Although sometimes the second page already gets loaded without scrolling.

Any suggestion on how to work around this while there is no fix available would be appreciated.

### Environment

Stackblitz

### Reproducer

https://stackblitz.com/edit/5pyzwj?file=src%2Fapp%2Fdemo%2Ftable-virtual-scroll-lazy-demo.ts

### Angular version

16.0.0 & 16.1.3

### PrimeNG version

16.0.2

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.20.0

### Steps to reproduce the behavior

1. Open Stackblitz
2. Wait for first 3 load events to complete
3. Scroll down

In case you get an initial timing that works:
1. Use a filter (i.e. Color)
2. Wait a few seconds and check if second page was not loaded yet
3. Scroll down

### Expected behavior

Pages should always be loaded as you scroll down.

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
